test/distributed/test_distributed.py

```diff
@@ -247,7 +247,7 @@ def init(cls):
             os.unlink(os.path.join(barrier_dir, f_name))
 
     @classmethod
-    def sync(cls, wait_for=None, timeout=5):
+    def sync(cls, wait_for=None, timeout=10):
```

Comment:
What is the reason for this change?

Reply:
The subtests `TestDistBackend.test_backend_group` and `TestDistBackend.test_backend_full_group`  were recently skipped for the ROCm platform due to ROCm CI failures.  The root cause was a timeout:

```
Process process 0:
Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/pytorch/test/distributed/test_distributed.py", line 2212, in _run
    self._barrier()
  File "/pytorch/test/distributed/test_distributed.py", line 280, in _barrier
    Barrier.sync(*args, **kwargs)
  File "/pytorch/test/distributed/test_distributed.py", line 274, in sync
    raise RuntimeError("barrier timeout")
RuntimeError: barrier timeout
FAIL
```

Reply:
I see, can we run the test in asan mode (usually the slowest) about 100 times locally to make sure the test isn't flaky with the new timeout?

Reply:
Is asan mode supported on HIP/CUDA?  Only reference I find to asan mode  is [here](https://github.com/pytorch/pytorch/blob/d6b51e4adfb34a177c89ed966e245abd49b14833/.jenkins/pytorch/build-asan.sh#L36), and it seems `USE_CUDA=0` for asan builds.